src/WebJobs.Extensions.DurableTask/ManagedIdentityTokenSource.cs

```diff
@@ -34,13 +36,19 @@ public ManagedIdentityTokenSource(string resource)
         [JsonProperty("resource")]
         public string Resource { get; }
 
+        /// <summary>
+        /// The azure credential options that a user can configure when authenticating.
+        /// </summary>
+        [JsonProperty("options")]
+        public DefaultAzureCredentialOptions Options { get; }
+
         /// <inheritdoc/>
         public async Task<string> GetTokenAsync()
         {
             var scopes = new string[] { this.Resource };
             TokenRequestContext context = new TokenRequestContext(scopes);
 
-            DefaultAzureCredential defaultCredential = new DefaultAzureCredential();
+            DefaultAzureCredential defaultCredential = new DefaultAzureCredential(this.Options);
```

Comment:
Is a `null` value allowed for this constructor? It might be safer to do something like this:
```suggestion
            DefaultAzureCredential defaultCredential = this.Options != null ?
                new DefaultAzureCredential(this.Options) :
                new DefaultAzureCredential();
```